### PR TITLE
Em/inversioncleanup

### PR DESCRIPTION
**Pull request by lheagy**
- let survey have both std and eps (percentage and a floor, respectively)
- if survey has std and eps, use these in the definition of the data misfit
- use `u` instead of `f` for fields in Jvec, Jtvec to be consistent with the rest of SimPEG
- add a 1D FDEM example inversion
- use survey.getSyntheticData in both FDEM and TDEM examples
- added a transpose property to the zero class (with test) 
- from SimPEG.EM import mu_0

### CODE REVIEW

File: SimPEG/DataMisfit.py [hunk 1]
```diff
@@ -103,10 +89,18 @@ def Wd(self):
...
             survey = self.survey
-            eps = np.linalg.norm(Utils.mkvc(survey.dobs),2)*1e-5
-            self._Wd = Utils.sdiag(1/(abs(survey.dobs)*survey.std+eps))
+
+            if getattr(survey,'std', None) is None:
+                print 'SimPEG.DataMisfit.l2_DataMisfit assigning default std of 5%'
+                survey.std = 0.05
+
+            if getattr(survey, 'eps', None) is None:
+                print 'SimPEG.DataMisfit.l2_DataMisfit assigning default eps of 1e-5 * ||dobs||'
```
Comment (rowanc1) [inline SimPEG/DataMisfit.py]: yay comments!

File: SimPEG/Survey.py [hunk 2]
```diff
@@ -205,6 +205,7 @@ class BaseSurvey(object):
     __metaclass__ = Utils.SimPEGMetaClass
 
     std = None       #: Estimated Standard Deviations
+    eps = None       #: Estimated Noise Floor
```
Comment (rowanc1) [inline SimPEG/Survey.py]: I suppose these would be better in the data class? But here is a good place for them now.

**Comment by lheagy** [comment]
@grosenkj , @rowanc1 , @sgkang : can you guys have a look at what I did, in particular wrt the survey having a std and eps. I think eventually this should be tied to the data, but for now, at least it will be used when we define Wd.

**Comment by grosenkj** [comment]
It makes more sense to me to have std and eps in the Data class, rather then survey. I like the suggestions in issue #202, since it is both the floor and uncert has to do with the data not survey per say. 

Would it make sense to restrict survey.dobs to be a SimPEG data class, rather than a numpy array?

**Comment by lheagy** [comment]
@grosenkj : agreed. I think we want to address that in a separate pr though? That one will take a little bit of thought

**Comment by rowanc1** [comment]
This looks good to me.

**Comment by rowanc1** [comment]
(as in, there is a lot more thought involved in the data/dobs/std/eps conversation, outside the scope of this PR)

**Pull request merged**